traffic_ops/traffic_ops_golang/invalidationjobs/invalidationjobs.go

```diff
@@ -0,0 +1,928 @@
+		&job.StartTime)
+	if err != nil {
+		sysErr = fmt.Errorf("Updating a job: %v", err)
+		errCode = http.StatusInternalServerError
+		api.HandleErr(w, r, inf.Tx.Tx, errCode, nil, sysErr)
+		return
+	}
+
+	if err = setRevalFlags(*job.DeliveryService, inf.Tx.Tx); err != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, http.StatusInternalServerError, nil, fmt.Errorf("Setting reval flags: %v", err))
+		return
+	}
+
+	response := apiResponse{
+		[]tc.Alert{
+			tc.Alert{
+				"Content invalidation job updated",
+				tc.SuccessLevel.String(),
+			},
+		},
+		job,
+	}
+
+	resp, err := json.Marshal(response)
+	if err != nil {
+		sysErr = fmt.Errorf("encoding response: %v", err)
+		errCode = http.StatusInternalServerError
+		api.HandleErr(w, r, inf.Tx.Tx, errCode, nil, sysErr)
+	} else {
+		w.Header().Set(http.CanonicalHeaderKey("content-type"), tc.ApplicationJson)
+		w.Write(resp)
+		w.Write([]byte{'\n'})
+	}
+
+	api.CreateChangeLogRawTx(api.ApiChange, api.Updated+"content invalidation job: #"+strconv.FormatUint(*job.ID, 10), inf.User, inf.Tx.Tx)
+}
+
+// Used by DELETE requests to `/jobs`, deletes an existing content invalidation job
+func Delete(w http.ResponseWriter, r *http.Request) {
+	inf, userErr, sysErr, errCode := api.NewInfo(r, []string{"id"}, []string{"id"})
+	if userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, errCode, userErr, sysErr)
+		return
+	}
+	defer inf.Close()
+
+	var dsid uint
+	var createdBy uint
+	row := inf.Tx.Tx.QueryRow(`SELECT job_deliveryservice, job_user FROM job WHERE id=$1`, inf.Params["id"])
+	if err := row.Scan(&dsid, &createdBy); err != nil {
+		if err == sql.ErrNoRows {
+			userErr = fmt.Errorf("No job by id '%s'!", inf.Params["id"])
+			errCode = http.StatusNotFound
+		} else {
+			sysErr = fmt.Errorf("Getting info for job #%s: %v", inf.Params["id"], err)
+			errCode = http.StatusInternalServerError
+		}
+		api.HandleErr(w, r, inf.Tx.Tx, errCode, userErr, sysErr)
+		return
+	}
+
+	if userErr, sysErr, errCode = IsUserAuthorizedToModifyDSID(inf, dsid); userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, errCode, userErr, sysErr)
+		return
+	}
+
+	if userErr, sysErr, errCode = IsUserAuthorizedToModifyJobsMadeByUserID(inf, createdBy); userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, errCode, userErr, sysErr)
+		return
+	}
+
+	result := tc.InvalidationJob{}
+	row = inf.Tx.Tx.QueryRow(deleteQuery, inf.Params["id"])
+	err := row.Scan(&result.AssetURL,
+		&result.CreatedBy,
+		&result.DeliveryService,
+		&result.ID,
+		&result.Keyword,
+		&result.Parameters,
+		&result.StartTime)
+	if err != nil {
+		sysErr = fmt.Errorf("deleting job #%s: %v", inf.Params["id"], err)
+		errCode = http.StatusInternalServerError
+		api.HandleErr(w, r, inf.Tx.Tx, errCode, nil, sysErr)
+		return
+	}
+
+	if err = setRevalFlags(dsid, inf.Tx.Tx); err != nil {
+		userErr = errors.New("Something wicked happened... Please queue server updates (or set reval_pending=true) manually to avoid problems!")
```

Comment:
unsure if we should use the verbiage _Something wicked happened_ to the user  :P